services/user/Webmail/src/Webmail.cpp

```diff
@@ -0,0 +1,178 @@
+      check(initRecord.has_value(), Errors::uninitialized);
+   }
+}
+
+void Webmail::init()
+{
+   auto initTable = Tables().open<InitTable>();
+   auto init      = (initTable.get(SingletonKey{}));
+   check(not init.has_value(), Errors::alreadyInit);
+   initTable.put(InitializedRecord{});
+
+   // Configure manualDebit for self on Token and NFT
+   to<Tokens>().setUserConf("manualDebit"_m, true);
+   to<Nft>().setUserConf("manualDebit"_m, true);
+
+   // Register http server
+   to<SystemService::HttpServer>().registerServer(Webmail::service);
+
+   // Events/indices
+   to<EventIndex>().setSchema(ServiceSchema::make<Webmail>());
+   to<EventIndex>().addIndex(DbId::historyEvent, Webmail::service, "sent"_m, 0);
+   to<EventIndex>().addIndex(DbId::historyEvent, Webmail::service, "sent"_m, 1);
+}
+
+void Webmail::send(psibase::AccountNumber receiver,
+                   const std::string&     subject,
+                   const std::string&     body)
+{
+   check(to<SystemService::Accounts>().exists(receiver), "target receiver account does not exist");
+   emit().history().sent(getSender(), receiver, subject, body);
+}
+
+// Supported REST query:
+//    `GET /messages?sender=alice&receiver=bob`
+// Either sender or receiver (or both) must be specified.
+optional<HttpReply> serveRestApi(const HttpRequest& request)
+{
+   if (request.method == "GET")
+   {
+      if (!request.target.starts_with("/messages"))
+      {
+         return nullopt;
+      }
+
+      size_t queryStart = request.target.find('?');
+      if (queryStart == string::npos)
+      {
+         return nullopt;
+      }
+
+      string query  = request.target.substr(queryStart + 1);
+      auto   params = parseQuery(query);
+
+      string s, r;
+      if (params.count("sender") != 0)
+      {
+         auto user = params.at("sender");
+         if (!validateUser(user))
+         {
+            return nullopt;
+         }
+         s = "sender = '" + user + "'";
+      }
+      if (params.count("receiver") != 0)
+      {
+         auto user = params.at("receiver");
+         if (!validateUser(user))
+         {
+            return nullopt;
+         }
+         r = "receiver = '" + user + "'";
+      }
+      if (s.empty() && r.empty())
+      {
+         return nullopt;
+      }
+
+      string where = "WHERE " + s;
+      if (!s.empty() && !r.empty())
+      {
+         where += " AND " + r;
+      }
+      else if (!r.empty())
+      {
+         where = "WHERE " + r;
+      }
+
+      return to<REvents>().serveSys(makeQuery(
+          request, "SELECT * FROM \"history.webmail.sent\" " + where + " ORDER BY ROWID"));
+   }
+   return nullopt;
+}
+
+optional<HttpReply> Webmail::serveSys(HttpRequest request)
+{
+   if (auto result = serveContent(request, Tables{}))
+      return result;
+   if (auto result = serveSimpleUI<Webmail, true>(request))
+      return result;
+   if (auto result = serveRestApi(request))
+      return result;
```

Comment:
I don't know when/where these might overlap, but wouldn't you want the serveSimpleUI to come last, given it's basically a fallback?

Reply:
The UI part is a fallback, so it should at least come after `serveContent`